typescript/sdk/src/token/EvmERC20WarpRouteReader.ts

```diff
@@ -126,6 +128,27 @@ export class EvmERC20WarpRouteReader extends HyperlaneReader {
       try {
         const warpRoute = factory.connect(warpRouteAddress, this.provider);
         await warpRoute[method]();
+
+        // If its a collateral, it could be an xerc20 token
+        if (tokenType === TokenType.collateral) {
+          // Try if its HypXERC20Lockbox
+          const hypXERC20Lockbox = HypXERC20Lockbox__factory.connect(
+            warpRouteAddress,
+            this.provider,
+          );
+          try {
```

Comment:
Why can’t these be added to the above contractTypes, similar to what we do for collateralVaults?

Reply:
Ah yes, good point

Reply:
Unfortunately, doesn't work for the HypXERC20 one, we have to go into the token